backend/analytics_server/mhq/service/code/sync/etl_github_handler.py

```diff
@@ -172,6 +173,10 @@ def process_pr(
         pr_events_model_list: List[PullRequestEvent] = self._to_pr_events(
             reviews, pr_model, pr_event_model_list
         )
+        pr_timeline_events = self._api.get_pr_timeline(org_name, org_repo, pr.number)
+        pr_earliest_ready_for_review = self.get_first_ready_for_review_event(
+            pr_timeline_events
+        )
```

Comment:
If we fetch this timeline in memory and extract this singular event for `READY_FOR_REVIEW` it changes our logic as a whole while leaving no trace of the specific `PullRequestEvent` in the DB as it was never saved. This leads to a bunch of issues:
- Makes debugging harder: finding out why and how a CycleTime number came to be.
- Makes data migrations harder: changing logic for CT will need refetching of the data from Cloud and DB data cant be used. 
- Timeline is a superset of PR events. So calling both APIs leads to redundant API calls.

Reply:
Ideal Approach:
- Use PR Timeline events API and adapt those to PullRequestEvent model. Include `READY_FOR_REVIEW` in the saved events and this should make things uniform without having to rely on a `datetime` object